src/instructlab/eval/mmlu.py

```diff
@@ -30,10 +36,49 @@ def run(self) -> tuple:
             overall_score       MMLU score for the overall model evaluation
             individual_scores   Individual MMLU score for each task
         """
-        individual_scores: dict[str, float] = {}
-        overall_score: float = 0.0
+        #TODO: make this a parameter for class?
+        os.environ['TOKENIZERS_PARALLELISM'] = 'true'
+
+        individual_scores: dict = {}
+        agg_score: float = 0.0
+        model_args = "pretrained=" + self.model_path + ",dtype=" + self.model_dtype
+
+        mmlu_output = simple_evaluate(
+            model="hf",
+            model_args=model_args,
+            tasks=self.tasks,
+            num_fewshot=self.few_shots,
+            batch_size=self.batch_size
+        )
+
+        results = mmlu_output["results"]
+
+        for task in self.tasks:
+            mmlu_res = results[task]
+            agg_score += float(mmlu_res['acc,none'])
+            individual_scores[task] = {}
+            individual_scores[task]['score'] = float(mmlu_res['acc,none'])
+            individual_scores[task]['stderr'] = float(mmlu_res['acc_stderr,none'])
+
+        overall_score = float(agg_score/len(self.tasks))
         return overall_score, individual_scores
+    
+
+############# Testing Code Follows ##############
+def main():
+    # Path to the granite model in the aliryan vm on AWS
+    model_path = "/home/ec2-user/instructlab/models/instructlab/granite-7b-lab"
+    #TODO: all 57 tasks need to be parameterized possibly by CLI
+    tasks = ["mmlu_abstract_algebra","mmlu_anatomy","mmlu_astronomy"]
+    dtype = "float16"
+    mmlu = MMLU_Evaluator(model_path, tasks, dtype, 2, 5)
+    overall_score, individual_scores = mmlu.run()
+    print(overall_score)
+    print(individual_scores)
 
+if __name__ == "__main__":
+    main()
```

Comment:
You might find it convenient to use `fire.Fire` for some of this in-lib testing code.

e.g.

```
import fire

def main(
    model_path: str = "instructlab/granite-7b-lab", 
    tasks: list[str] = None, 
    dtype: str = "bf16"):
    pass

if __name__ == "__main__":
    fire.Fire(main)
```

then you can invoke the script with

`python3 -m instructlab.train.mmlu --model_path <something>`

fire handles parsing args and passing them to the function / module that you wrap.

Reply:
Thank you - I'm going to skip this for now since it has already been tested. I appreciate the suggestion